src/backend/gl/src/native.rs

```diff
@@ -38,6 +43,79 @@ impl Fence {
     }
 }
 
+#[derive(Copy, Clone, Debug, Eq, Hash, PartialEq)]
+pub(crate) enum BindingTypes {
+    UniformBuffers,
+    StorageBuffers,
+    StorageImages,
+    SampledImages,
+    SeparateImages,
+    SeparateSamplers,
+    SubpassInputs,
+}
+
+#[derive(Clone, Debug)]
+pub(crate) struct DescRemapData {
+    bindings: HashMap<BindingTypes, HashMap<SetID, HashMap<BindingID, BindingID>>>,
```

Comment:
this could benefit from squashing all 3 hashmaps into a single one:
```rust
bindings: HashMap<(pso::DescriptorType, pso::DescriptorSetIndex, pso::DescriptorBinding), pso::DescriptorBinding>,
```